synapseclient/core/download/download_functions.py

```diff
@@ -719,21 +735,69 @@ def _ftp_report_hook(
                 else None
             )
 
-            response = with_retry(
-                lambda url=url, range_header=range_header, auth=auth: client._requests_session.get(
-                    url=url,
-                    headers=client._generate_headers(range_header),
-                    stream=True,
-                    allow_redirects=False,
-                    auth=auth,
-                ),
-                verbose=client.debug,
-                **STANDARD_RETRY_PARAMS,
-            )
             try:
+                url_has_expiration = "Expires" in urllib_urlparse.urlparse(url).query
```

Comment:
Just a check here : is it worthwhile to put this in a helper function or is that just unnecessary overhead? (Since it's only used twice)

Reply:
I had the same thought but figured it wasn't worth it because of the changes to tests and new tests that would be needed.

Reply:
The one thought I had is whether this would be easier to test as a unit test.  That said - i think the test cases you added does cover for these scenarios - it may become an extra step to change code in both areas if things need to be updated.